src/main.ts

```diff
@@ -0,0 +1,161 @@
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import * as fs from 'fs';
+import { request } from '@octokit/request';
+import { Octokit } from "@octokit/core";
+fs.readdir('./', (err, files) => {
+  if (err)
+    console.log(err);
+  else {
+
+    //Check for example and Contribution in README
+    readmeChecks(files);
+
+    //Check for CODEOWNERS file in .github folder
+    codeownerCheck();
+
+    //Check if nodemodules folder is present in master branch for typescript action
+    nodemodulesCheck();
+
+    //check for branch permissions in main
+    start('main');
+
+    //check for nodemodules folder in releases/*
+    releasesfunc();
+    
+    }
+  })
+
+function getExtension(filename: string) {
+  return filename.substring(filename.lastIndexOf('.')+1, filename.length)   
+}
+
+function readmeChecks(files: string[]){
+
+  const includesReadme = files.includes('README.md');
+  if(includesReadme){
+    console.log("README file is present");
+    fs.readFile('./README.md', function (err, data) {
+    //check example
+    if(data.includes('Example'))
+      console.log("Example workflow is present in README");
+    else
+      core.setFailed("Please add Example workflow in README");
+                
+      //check contribution
+    if(data.includes('Contribution'))
+      console.log("Contribution is present in README");
+                 
```

Comment:
Use braces for if-else block
and leave blank lines only where necessary

Ensure proper formatting